ttnn/cpp/ttnn/graph/graph_processor.cpp

```diff
@@ -307,8 +307,27 @@ int GraphProcessor::add_tensor(const Tensor& t) {
     return tensor_counter;
 }
 
+namespace detail
+{
+    // hash buffer because address is not unique
+    inline size_t hash_object(const tt::tt_metal::Buffer* buffer) noexcept {
```

Comment:
Must accept a const ref or check for nullptr

Reply:
@dmakoviichuk-tt aby objections against hashing a pointer here?

hashing buffer address won’t make it because it might be a reallocated slot.

Reply:
this wont work anyway. there is always an edge case that would mess our lifes. I'll add unique id to buffer